perma_web/perma/tests/test_views_auth.py

```diff
@@ -1,159 +1,149 @@
-        self.assertContains(response, 'Too Many Attempts', status_code=403)
-        self.assertContains(response, 'Reset my password', status_code=403)
-        aa = AccessAttempt.objects.get(username=self.email)
-        self.assertEqual(aa.failures_since_start, 2)
-
-        # get the reset password email/link
-        self.client.post(reverse('password_reset'), {"email": self.email}, secure=True)
-        message = mail.outbox[0]
-        reset_url = next(line for line in message.body.rstrip().split("\n") if line.startswith('http'))
-
-        # go through with the reset
-        response = self.client.get(reset_url, follow=True, secure=True)
-        post_url = response.redirect_chain[0][0]
-        response = self.client.post(post_url, {'new_password1': self.new_password, 'new_password2': self.new_password}, follow=True, secure=True)
-        self.assertContains(response, 'Your password has been set')
-        self.assertFalse(AccessAttempt.objects.filter(username=self.email).exists())
-
-        # verify you get the normal form errors, not the lockout page,
-        # if you fail again
-        response = self.attempt_login(self.email, self.wrong_password, expect_success=False)
-        self.assertContains(response, 'class="field-error"', status_code=200)
-
-        # verify you CAN login with the new password
-        self.log_in_user(user=self.email, password=self.new_password)
+
+from conftest import TEST_USER_PASSWORD, randomize_capitalization, submit_form, log_in_user
+
+
+def attempt_login(perma_client, username, password, expect_success=True):
+    assert '_auth_user_id' not in perma_client.session
+
+    response = perma_client.post(reverse('user_management_limited_login'),
+                                 {'username': username,
+                                  'password': password})
+    if expect_success:
+        assert response.status_code == 302
+        assert 'login' not in response['Location']
+        assert '_auth_user_id' in perma_client.session
+    else:
+        assert '_auth_user_id' not in perma_client.session
+    return response
+
+def test_login(perma_client, link_user):
+    """
+    Test the login form
+    We should get redirected to the create page
+    """
+    # Login through our form and make sure we get redirected to our create page,
+    # no matter how the email address is capitalized
+    assert LinkUser.objects.filter(email__iexact=link_user.email).count() == 1
+
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.upper(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.title(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, randomize_capitalization(link_user.email), TEST_USER_PASSWORD)
+
+def test_deactived_user_login(perma_client, deactivated_registrar_user):
```

Comment:
Just curious: any particular reason you went with a registrar user instead of a standard user for the deactivated/and not-yet-activated login tests? (Which is fine)

Reply:
Wow I have no idea. I for some reason thought these were registrar users. I'll change for simplicity's sake.